staging/src/k8s.io/client-go/discovery/discovery_client.go

```diff
@@ -88,7 +88,10 @@ type ServerResourcesInterface interface {
 	// ServerResourcesForGroupVersion returns the supported resources for a group and version.
 	ServerResourcesForGroupVersion(groupVersion string) (*metav1.APIResourceList, error)
 	// ServerResources returns the supported resources for all groups and versions.
+	// Deprecated: use ServerGroupsAndResources instead.
 	ServerResources() ([]*metav1.APIResourceList, error)
+	// ServerResources returns the supported groups and resources for all groups and versions.
```

Comment:
not new, but this interface should document the possibility of partial results, and what error will be used to communicate that

Reply:
done